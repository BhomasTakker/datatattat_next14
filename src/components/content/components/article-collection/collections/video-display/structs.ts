export const enum PlayerCollectionVariant {
	HorizontalScroll = "HorizontalScroll",
	VerticalScroll = "VerticalScroll",
}

export const enum PlayerSourceTypes {
	Youtube = "video/youtube",
	Vimeo = "video/vimeo",
	Mp4 = "video/mp4",
}

export enum PlayerAutoplayOptions {
	Play = "play",
	Muted = "muted",
	Any = "any",
}
