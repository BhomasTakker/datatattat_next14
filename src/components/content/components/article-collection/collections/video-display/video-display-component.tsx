"use client";

import { CollectionItem } from "@/types/data-structures/collection/item/item";
import { VideoPlayerContainer } from "../../../video-player/video-player-container";
import { Interaction } from "../../article/interaction/interactions";
import { InteractionsOptions } from "../../article/interaction/interactions-map";

import styles from "./video-display.module.scss";
import { useRef } from "react";
import Player from "video.js/dist/types/player";
import {
	PlayerAutoplayOptions,
	PlayerCollectionVariant,
	PlayerSourceTypes,
} from "./structs";
import { InViewComponent } from "@/components/ui/in-view/in-view";
import { articleMetaLoader, articleRenderer, articleTemplate } from "../utils";
import { WithData } from "@/components/ui/with-data/with-data";
import { BASE_PLAYER_OPTIONS } from "../../../video-player/options";

type VideoDisplayComponentProps = {
	articles: CollectionItem[];

	variant: PlayerCollectionVariant;
	sourceType: PlayerSourceTypes;
	autoplay: boolean;
};

export const VideoDisplayComponent = ({
	articles = [],
	variant,
	sourceType, // use youtube as an id and then do whatever required for each option
	autoplay,
}: VideoDisplayComponentProps) => {
	const playerRef = useRef<Player>(null);
	const firstArticle = articles[0];
	const src = firstArticle?.src;
	const template = articleTemplate(styles);

	if (!firstArticle) {
		return null;
	}

	// These in a separate file
	// We will ultimately need to pass or reference sourcetype
	// based on item as we may want variant sources in the same collection
	const onClickHnd = (item: CollectionItem) => {
		const video = playerRef.current;
		if (!video) {
			return;
		}
		// This works to effectively reset the error - if there is an error
		// @ts-expect-error null works but flags error / undefine ddoes not work...
		video.error(null);
		video.autoplay(true);
		video.src({
			src: item.src,
			type: sourceType,
		});
		video.poster(item.avatar?.src);
	};

	const options = {
		...BASE_PLAYER_OPTIONS,
		autoplay: autoplay ? PlayerAutoplayOptions.Any : false,
		poster: firstArticle.avatar?.src || "",
		sources: [
			{
				src,
				type: sourceType,
			},
		],
	};

	const containerClassName = `container${variant}`;
	const containerClass = styles[containerClassName];

	// We want to create the video initialisation object here?
	// If we are passing in event handlers then we may well want to

	return (
		// class variant
		<div className={containerClass} data-testid={variant}>
			<div className={styles.videoPlayer} data-testid="video-player-container">
				<VideoPlayerContainer
					styles={styles}
					playerRef={playerRef}
					options={options}
				/>
			</div>
			<ul className={styles.articles} data-testid="articles-list">
				{articles.map((item) => (
					<li key={item.src}>
						<InViewComponent
							options={{
								threshold: 0,
								triggerOnce: true,
							}}
							template={<div className={styles.template} />}
						>
							{/* Should be part of a 'wider' Article Component  */}
							<Interaction
								key={item.title}
								type={InteractionsOptions.Click}
								onClick={() => onClickHnd(item)}
							>
								<WithData
									getter={articleMetaLoader(item)}
									callback={articleRenderer(styles)}
									template={template}
								/>
							</Interaction>
						</InViewComponent>
					</li>
				))}
			</ul>
		</div>
	);
};
