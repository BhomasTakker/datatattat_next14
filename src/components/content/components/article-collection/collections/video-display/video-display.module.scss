@use "../../article/mixins/" as article;
@use "../../../../../../scss/abstracts/units/" as *;
@use "../../../../../../scss/mixins/ui/" as ui;

.root {
	.containerHorizontalScroll {
		.videoPlayer {
			display: flex;
			justify-content: center;
			// can add width here
			padding-bottom: 1.5rem;
		}

		ul {
			display: flex;
			gap: $rem_4;
			@include ui.scrollXNoScroller();
			@include article.articleCardTopDown($rem_14, $rem_14);
			@media screen and (min-width: $screenWidthSm) {
				@include article.articleCardTopDown($rem_15, $rem_14);
			}
			@media screen and (min-width: $screenWidthMd) {
				@include article.articleCardTopDown($rem_16, $rem_14);
			}
		}
	}

	.containerVerticalScroll {
		.videoPlayer {
			display: flex;
			justify-content: center;
			// can add width here
			padding-bottom: 1.5rem;
		}

		.articles {
			display: flex;
			gap: $rem_4;
			@include ui.scrollXNoScroller();
			@include article.articleCardTopDown($rem_14, $rem_14);
			@media screen and (min-width: $screenWidthSm) {
				@include article.articleCardTopDown($rem_15, $rem_14);
			}
			@media screen and (min-width: $screenWidthMd) {
				@include article.articleCardTopDown($rem_16, $rem_14);
			}
		}

		@media screen and (min-width: $screenWidthXl) {
			display: grid;
			grid-template-columns: 2fr 1fr;
			gap: $rem_5;

			.videoPlayer {
				padding-bottom: 0;
			}

			.articles {
				height: 0;
				min-height: 100%;

				display: flex;
				flex-direction: column;
				gap: $rem_2;

				@include ui.scrollXNoScroller();

				@media screen and (min-width: $screenWidthXl) {
					.article {
						all: initial;
						* {
							all: unset;
						}
					}
					@include article.articleCardLeftRight($rem_16, $rem_8);
				}
			}
		}
	}
}
