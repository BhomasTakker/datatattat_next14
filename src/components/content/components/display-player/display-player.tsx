import { PageComponent } from "@/types/page";

export const DisplayPlayer = ({ component }: { component: PageComponent }) => {
	console.log({ component });
	return <div>Display Player</div>;
};
