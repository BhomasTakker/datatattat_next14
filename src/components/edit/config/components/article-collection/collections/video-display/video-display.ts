import {
	PlayerCollectionVariant,
	PlayerSourceTypes,
} from "@/components/content/components/article-collection/collections/video-display/structs";
import { EditInputs } from "@/components/edit/inputs/inputs";
import { InputListProps } from "@/types/edit/inputs/inputs";

// We may need video source

export const VIDEO_DISPLAY: InputListProps = {
	id: "videoDisplay",
	type: EditInputs.inputList,
	label: "Video Display",
	inputs: [
		{
			id: "videoDisplayTitle",
			type: EditInputs.title,
			title: "Video Display",
		},
		{
			id: "videoDisplayDescription",
			type: EditInputs.description,
			text: "This component displays a collection of videos in a vertical or horizontal scroll. Please note currently only youtube videos are available.",
		},
		{
			id: "variant",
			type: EditInputs.select,
			label: "Video Display Variant",
			defaultValue: PlayerCollectionVariant.VerticalScroll,
			required: true,
			options: [
				PlayerCollectionVariant.VerticalScroll,
				PlayerCollectionVariant.HorizontalScroll,
			],
		},
		{
			id: "sourceType",
			type: EditInputs.select,
			label: "Video Source Type",
			defaultValue: PlayerSourceTypes.Youtube,
			required: true,
			options: [
				PlayerSourceTypes.Youtube,
				// PlayerSourceTypes.Vimeo,
				// PlayerSourceTypes.Mp4,
			],
		},
		{
			id: "autoplay",
			type: EditInputs.switch,
			label: "Autoplay",
			defaultChecked: false,
			// defaultValue: false,
		},
	],
};
