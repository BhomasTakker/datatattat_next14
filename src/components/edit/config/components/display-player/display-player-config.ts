import { PlayerSourceTypes } from "@/components/content/components/article-collection/collections/video-display/structs";
import { EditInputs } from "@/components/edit/inputs/inputs";
import { InputListProps } from "@/types/edit/inputs/inputs";

export const DISPLAY_PLAYER_CONFIG: InputListProps = {
	id: "displayPlayerComponent",
	type: EditInputs.inputList,
	label: "Display Player Component",
	inputs: [
		{
			id: "displayPlayerComponentTitle",
			type: EditInputs.title,
			title: "Display Player Component",
		},
		{
			id: "sourceType",
			type: EditInputs.select,
			label: "Video Source Type",
			defaultValue: PlayerSourceTypes.Youtube,
			required: true,
			options: [
				PlayerSourceTypes.Youtube,
				// PlayerSourceTypes.Vimeo,
				PlayerSourceTypes.Mp4,
			],
		},
		{
			id: "src",
			type: EditInputs.text,
			label: "Video Source URL",
			// placeholder: 'Enter the video source URL',
			required: true,
			// defaultValue: 'https://www.youtube.com/watch?v=example', // Example URL
		},
		{
			id: "poster",
			type: EditInputs.text,
			label: "Poster Image URL",
			// placeholder: "Enter the poster image URL (optional)",
			required: false,
			defaultValue: "", // Optional poster image URL
		},
		// Update with - something like filling out a playlist
		// getWithConfig({ options: [QueryOptions.NONE, QueryOptions.OEMBED] }),
	],
};
