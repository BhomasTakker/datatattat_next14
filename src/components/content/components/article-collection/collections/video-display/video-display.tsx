import { CollectionItem } from "@/types/data-structures/collection/item/item";
import styles from "./video-display.module.scss";
import { VideoDisplayComponent } from "./video-display-component";

const renderMethod = (articles: CollectionItem[]) => {
	if (!articles || articles.length === 0) return null;
	// Will be video component horizontal scroll or something
	return <VideoDisplayComponent articles={articles} />;
};

export const videoDisplay = {
	styles: styles,
	renderMethod: renderMethod,
};
