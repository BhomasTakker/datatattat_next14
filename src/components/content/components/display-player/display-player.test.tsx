import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import { VideoDisplayComponent, DisplayPlayer } from "./display-player";
import { PlayerSourceTypes } from "../article-collection/collections/video-display/structs";
import type { ComponentProps } from "@/types/component";

// Mock the VideoPlayerContainer component
jest.mock("../video-player/video-player-container", () => ({
	VideoPlayerContainer: ({ styles, playerRef, options }: any) => (
		<div
			data-testid="video-player-container"
			data-options={JSON.stringify(options)}
			data-styles={styles?.root || ""}
		>
			Mock Video Player Container
		</div>
	),
}));

// Mock the BASE_PLAYER_OPTIONS
jest.mock("../video-player/options", () => ({
	BASE_PLAYER_OPTIONS: {
		autoplay: false,
		aspectRatio: "16:9",
		controls: true,
		fill: true,
		fluid: true,
		responsive: true,
		poster: "",
		playbackRates: [0.5, 1, 1.5, 2],
		width: undefined,
		height: undefined,
		preload: "auto",
		techOrder: ["html5", "youtube"],
		sources: [],
		audioOnlyMode: false,
		audioPosterMode: false,
	},
}));

// Mock the styles
jest.mock("./display-player.module.scss", () => ({
	root: "mock-display-player-root",
}));

describe("VideoDisplayComponent", () => {
	afterEach(() => {
		cleanup();
		jest.clearAllMocks();
	});

	describe("Basic Rendering", () => {
		it("renders the component with correct test id", () => {
			const options = {
				src: "https://example.com/video.mp4",
			};

			render(<VideoDisplayComponent options={options} />);

			expect(screen.getByTestId("display-player")).toBeInTheDocument();
			expect(screen.getByTestId("display-player")).toHaveClass(
				"mock-display-player-root"
			);
		});

		it("renders VideoPlayerContainer with correct props", () => {
			const options = {
				src: "https://example.com/video.mp4",
				sourceType: PlayerSourceTypes.Mp4,
				poster: "https://example.com/poster.jpg",
				audioOnlyMode: false,
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			expect(videoContainer).toBeInTheDocument();

			// Check if options are passed correctly
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.sources).toHaveLength(1);
			expect(optionsData.sources[0].src).toBe("https://example.com/video.mp4");
			expect(optionsData.sources[0].type).toBe("video/mp4");
			expect(optionsData.poster).toBe("https://example.com/poster.jpg");
			expect(optionsData.audioOnlyMode).toBe(false);
		});
	});

	describe("Source Configuration", () => {
		it("handles missing src prop", () => {
			const options = {
				src: "",
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.sources[0].src).toBe("");
			expect(optionsData.sources[0].type).toBeUndefined();
		});

		it("handles different source types", () => {
			const testCases = [
				{
					sourceType: PlayerSourceTypes.Youtube,
					expected: "video/youtube",
				},
				{
					sourceType: PlayerSourceTypes.Vimeo,
					expected: "video/vimeo",
				},
				{
					sourceType: PlayerSourceTypes.Mp4,
					expected: "video/mp4",
				},
				{
					sourceType: PlayerSourceTypes.Default,
					expected: "default",
				},
			];

			testCases.forEach(({ sourceType, expected }) => {
				const { unmount } = render(
					<VideoDisplayComponent
						options={{
							src: "https://example.com/video",
							sourceType,
						}}
					/>
				);

				const videoContainer = screen.getByTestId("video-player-container");
				const optionsData = JSON.parse(
					videoContainer.getAttribute("data-options") || "{}"
				);
				expect(optionsData.sources[0].type).toBe(expected);

				unmount();
			});
		});

		it("handles undefined sourceType", () => {
			const options = {
				src: "https://example.com/video.mp4",
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.sources[0].type).toBeUndefined();
		});
	});

	describe("Poster Configuration", () => {
		it("uses provided poster URL", () => {
			const options = {
				src: "https://example.com/video.mp4",
				poster: "https://example.com/poster.jpg",
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.poster).toBe("https://example.com/poster.jpg");
		});

		it("uses empty string when poster is not provided", () => {
			const options = {
				src: "https://example.com/video.mp4",
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.poster).toBe("");
		});
	});

	describe("Audio Mode Configuration", () => {
		it("handles audioOnlyMode true", () => {
			const options = {
				src: "https://example.com/audio.mp3",
				audioOnlyMode: true,
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.audioOnlyMode).toBe(true);
		});

		it("handles audioOnlyMode false", () => {
			const options = {
				src: "https://example.com/video.mp4",
				audioOnlyMode: false,
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.audioOnlyMode).toBe(false);
		});

		it("handles undefined audioOnlyMode", () => {
			const options = {
				src: "https://example.com/video.mp4",
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);
			expect(optionsData.audioOnlyMode).toBeUndefined();
		});
	});

	describe("Player Options Merging", () => {
		it("merges with BASE_PLAYER_OPTIONS correctly", () => {
			const options = {
				src: "https://example.com/video.mp4",
				sourceType: PlayerSourceTypes.Mp4,
				poster: "https://example.com/poster.jpg",
				audioOnlyMode: true,
			};

			render(<VideoDisplayComponent options={options} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);

			// Check that BASE_PLAYER_OPTIONS are merged
			expect(optionsData.autoplay).toBe(false);
			expect(optionsData.aspectRatio).toBe("16:9");
			expect(optionsData.controls).toBe(true);
			expect(optionsData.fill).toBe(true);
			expect(optionsData.fluid).toBe(true);
			expect(optionsData.responsive).toBe(true);
			expect(optionsData.playbackRates).toEqual([0.5, 1, 1.5, 2]);
			expect(optionsData.preload).toBe("auto");
			expect(optionsData.techOrder).toEqual(["html5", "youtube"]);

			// Check that provided options override base options
			expect(optionsData.poster).toBe("https://example.com/poster.jpg");
			expect(optionsData.audioOnlyMode).toBe(true);
		});
	});
});

describe("DisplayPlayer", () => {
	const createMockProps = (componentProps: any = {}): ComponentProps => ({
		component: {
			componentProps,
		} as any,
		dataObject: {
			data: {},
		},
	});

	afterEach(() => {
		cleanup();
		jest.clearAllMocks();
	});

	describe("Props Extraction", () => {
		it("extracts all props correctly from componentProps", () => {
			const componentProps = {
				src: "https://example.com/video.mp4",
				sourceType: PlayerSourceTypes.Youtube,
				poster: "https://example.com/poster.jpg",
				audioOnlyMode: true,
			};

			const props = createMockProps(componentProps);
			render(<DisplayPlayer {...props} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);

			expect(optionsData.sources[0].src).toBe("https://example.com/video.mp4");
			expect(optionsData.sources[0].type).toBe("video/youtube");
			expect(optionsData.poster).toBe("https://example.com/poster.jpg");
			expect(optionsData.audioOnlyMode).toBe(true);
		});

		it("handles partial componentProps", () => {
			const componentProps = {
				src: "https://example.com/video.mp4",
			};

			const props = createMockProps(componentProps);
			render(<DisplayPlayer {...props} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);

			expect(optionsData.sources[0].src).toBe("https://example.com/video.mp4");
			expect(optionsData.sources[0].type).toBeUndefined();
			expect(optionsData.poster).toBe(""); // BASE_PLAYER_OPTIONS default
			expect(optionsData.audioOnlyMode).toBeUndefined();
		});

		it("handles empty componentProps", () => {
			const props = createMockProps({});
			render(<DisplayPlayer {...props} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);

			expect(optionsData.sources[0].src).toBe("");
			expect(optionsData.sources[0].type).toBeUndefined();
			expect(optionsData.poster).toBe(""); // BASE_PLAYER_OPTIONS default
			expect(optionsData.audioOnlyMode).toBeUndefined();
		});

		it("uses default empty string for src when not provided", () => {
			const componentProps = {
				sourceType: PlayerSourceTypes.Mp4,
				poster: "https://example.com/poster.jpg",
				audioOnlyMode: false,
			};

			const props = createMockProps(componentProps);
			render(<DisplayPlayer {...props} />);

			const videoContainer = screen.getByTestId("video-player-container");
			const optionsData = JSON.parse(
				videoContainer.getAttribute("data-options") || "{}"
			);

			expect(optionsData.sources[0].src).toBe("");
			expect(optionsData.sources[0].type).toBe("video/mp4");
			expect(optionsData.poster).toBe("https://example.com/poster.jpg");
			expect(optionsData.audioOnlyMode).toBe(false);
		});
	});

	describe("Integration with VideoDisplayComponent", () => {
		it("passes correct options to VideoDisplayComponent", () => {
			const componentProps = {
				src: "https://example.com/test.mp4",
				sourceType: PlayerSourceTypes.Mp4,
				poster: "https://example.com/test-poster.jpg",
				audioOnlyMode: false,
			};

			const props = createMockProps(componentProps);
			render(<DisplayPlayer {...props} />);

			expect(screen.getByTestId("display-player")).toBeInTheDocument();
			expect(screen.getByTestId("video-player-container")).toBeInTheDocument();
		});
	});

	describe("Snaps", () => {
		it("renders DisplayPlayer unchanged", () => {
			const componentProps = {
				src: "https://example.com/video.mp4",
				sourceType: PlayerSourceTypes.Mp4,
				poster: "https://example.com/poster.jpg",
				audioOnlyMode: false,
			};

			const props = createMockProps(componentProps);
			const { container } = render(<DisplayPlayer {...props} />);
			expect(container).toMatchSnapshot();
		});

		it("renders VideoDisplayComponent unchanged", () => {
			const options = {
				src: "https://example.com/video.mp4",
				sourceType: PlayerSourceTypes.Mp4,
				poster: "https://example.com/poster.jpg",
				audioOnlyMode: false,
			};

			const { container } = render(<VideoDisplayComponent options={options} />);
			expect(container).toMatchSnapshot();
		});
	});
});
