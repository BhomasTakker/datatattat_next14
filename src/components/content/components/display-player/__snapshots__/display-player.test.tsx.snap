// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DisplayPlayer Snaps renders DisplayPlayer unchanged 1`] = `
<div>
  <div
    class="mock-display-player-root"
    data-testid="display-player"
  >
    <div
      data-options="{"autoplay":false,"aspectRatio":"16:9","controls":true,"fill":true,"fluid":true,"responsive":true,"poster":"https://example.com/poster.jpg","playbackRates":[0.5,1,1.5,2],"preload":"auto","techOrder":["html5","youtube"],"sources":[{"src":"https://example.com/video.mp4","type":"video/mp4"}],"audioOnlyMode":false,"audioPosterMode":false}"
      data-styles="mock-display-player-root"
      data-testid="video-player-container"
    >
      Mock Video Player Container
    </div>
  </div>
</div>
`;

exports[`DisplayPlayer Snaps renders VideoDisplayComponent unchanged 1`] = `
<div>
  <div
    class="mock-display-player-root"
    data-testid="display-player"
  >
    <div
      data-options="{"autoplay":false,"aspectRatio":"16:9","controls":true,"fill":true,"fluid":true,"responsive":true,"poster":"https://example.com/poster.jpg","playbackRates":[0.5,1,1.5,2],"preload":"auto","techOrder":["html5","youtube"],"sources":[{"src":"https://example.com/video.mp4","type":"video/mp4"}],"audioOnlyMode":false,"audioPosterMode":false}"
      data-styles="mock-display-player-root"
      data-testid="video-player-container"
    >
      Mock Video Player Container
    </div>
  </div>
</div>
`;
